sql/table.cc

```diff
@@ -7229,9 +7229,7 @@ Item *create_view_field(THD *thd, TABLE_LIST *view, Item **field_ref,
   {
     DBUG_RETURN(field);
   }
-  Name_resolution_context *context= (view->view ?
-                                     &view->view->first_select_lex()->context:
-                                     &thd->lex->first_select_lex()->context);
```

Comment:
this makes no sense.  If the view field is an outer reference and we give it a context of view->view->first_select_lex()->context we end up with the wrong result.  Example from main.outer_reference

select * from t6, v1 where t6a=v1a and
(
  select max(t4a) from t4 where t4c < v1b
) <> 0;

when we create the Item_direct_view_ref associated with v1b, it needs it's context to NOT be inside the view.